src/main/java/com/sproutsocial/nsq/Connection.java

```diff
@@ -197,12 +204,12 @@ private void checkHeartbeat() {
 
     @GuardedBy("this")
     protected void writeCommand(String cmd, Object param1, Object param2) throws IOException {
-        out.write(String.format("%s %s %s\n", cmd, param1, param2).getBytes(Util.US_ASCII));
+        out.write((cmd + " " + param1 + " " + param2 + "\n").getBytes(Util.US_ASCII));
     }
 
     @GuardedBy("this")
     protected void writeCommand(String cmd, Object param) throws IOException {
-        out.write(String.format("%s %s\n", cmd, param).getBytes(Util.US_ASCII));
+        out.write((cmd + " " +  param + "\n").getBytes(Util.US_ASCII));
```

Comment:
Can't justify this, but string.format is shockingly slow compared to StringBuffer. This line was actually showing up in a profile trace.